src/app/3d/qgs3dmaptoolpaintbrush.cpp

```diff
@@ -0,0 +1,165 @@
+  mHighlighterRubberBand->setEdgesEnabled( false );
+}
+
+void Qgs3DMapToolPaintBrush::deactivate()
+{
+  restart();
+  mSelectionRubberBand.reset();
+  mIsActive = false;
+  mCanvas->cameraController()->setInputHandlersEnabled( true );
+}
+
+QCursor Qgs3DMapToolPaintBrush::cursor() const
+{
+  return Qt::CrossCursor;
+}
+
+void Qgs3DMapToolPaintBrush::restart()
+{
+  mDragPositions.clear();
+  mHighlighterRubberBand->reset();
+  mIsClicked = false;
+}
+
+void Qgs3DMapToolPaintBrush::generateHighlightArea()
+{
+  const QgsGeometry searchSkeleton = QgsGeometry( new QgsLineString( mDragPositions ) );
+  const QgsGeometry searchGeometry = searchSkeleton.buffer( mSelectionRubberBand->width() / 2, 37 );
+  QgsPolygon *searchPolygon = qgsgeometry_cast<QgsPolygon *>( searchGeometry.constGet() );
+  Q_ASSERT( searchPolygon );
+  auto transform = [this]( const QgsPoint &point ) -> QgsPoint {
+    return Qgs3DUtils::screenPointToMapCoordinates( QPoint( point.x(), point.y() ), *mCanvas );
+  };
+  searchPolygon->addZValue( 0 );
+  searchPolygon->transformVertices( transform );
+  mHighlighterRubberBand->setGeometry( QgsGeometry( searchPolygon->clone() ) );
+}
+
+void Qgs3DMapToolPaintBrush::mousePressEvent( QMouseEvent *event )
+{
+  if ( event->button() == Qt::LeftButton && !mIsMoving )
+  {
+    mIsClicked = true;
+    mDragPositions.append( QgsPointXY( event->x(), event->y() ) );
+  }
+}
+
+void Qgs3DMapToolPaintBrush::mouseReleaseEvent( QMouseEvent *event )
+{
+  if ( mIsClicked && event->button() == Qt::LeftButton && !mIsMoving )
+  {
+    mDragPositions.append( QgsPointXY( event->x(), event->y() ) );
+    mHighlighterRubberBand->reset();
+    run();
+    mDragPositions.clear();
+  }
+  mIsClicked = false;
+}
+
+void Qgs3DMapToolPaintBrush::mouseMoveEvent( QMouseEvent *event )
+{
+  if ( mIsActive )
+  {
+    const QgsPoint newPos = Qgs3DUtils::screenPointToMapCoordinates( event->pos(), *mCanvas );
+    mSelectionRubberBand->moveLastPoint( newPos );
+
+    if ( mIsClicked && !mIsMoving )
+    {
+      mDragPositions.append( QgsPointXY( event->x(), event->y() ) );
+      generateHighlightArea();
+    }
+  }
+}
+
+void Qgs3DMapToolPaintBrush::mouseWheelEvent( QWheelEvent *event )
+{
+  // Moving horizontally or being in movement mode discards the event
+  if ( event->angleDelta().y() == 0 || mIsMoving )
+  {
+    event->accept();
+    return;
+  }
+
+  // Change the selection circle size. Moving the wheel forward (away) from the user makes
+  // the circle smaller
+  const QgsSettings settings;
+  const bool reverseZoom = settings.value( QStringLiteral( "qgis/reverse_wheel_zoom" ), false ).toBool();
+  const bool shrink = reverseZoom ? event->angleDelta().y() < 0 : event->angleDelta().y() > 0;
+  double zoomFactor = shrink ? 0.75 : 1.5;
+  // "Normal" mouse have an angle delta of 120, precision mouses provide data faster, in smaller steps
+  zoomFactor = 1.0 + ( zoomFactor - 1.0 ) / 120.0 * std::fabs( event->angleDelta().y() );
```

Comment:
let's simplify this - e.g.:
```
double zoomFactor = ( shrink ? 0.75 : 1.5 ) / 120.0 * std::fabs( event->angleDelta().y() );
```
(if I got the math right)

By the way - it seems like the zooming in and zooming out does not have the same speed - when shrinking, the change is 33% (1.0 / 0.75) and when expanding, the change is 50% (1.5 / 1.0)   ... but maybe I got the math wrong...